src/Panel/Controller/View/FileViewController.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace Kirby\Panel\Controller\View;
+
+use Kirby\Cms\File;
+use Kirby\Cms\Files;
+use Kirby\Cms\Find;
+use Kirby\Cms\ModelWithContent;
+use Kirby\Panel\Model;
+use Kirby\Panel\Ui\Button\ViewButtons;
+use Kirby\Panel\Ui\FilePreview;
+
+/**
+ * Controls the file view
+ *
+ * @package   Kirby Panel
+ * @author    Nico Hoffmann <nico@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ * @since     6.0.0
+ */
+class FileViewController extends ModelViewController
+{
+	/**
+	 * @var \Kirby\Cms\File $model
+	 */
+	protected ModelWithContent $model;
+
+	/**
+	 * @var \Kirby\Panel\File
+	 */
+	protected Model $panel;
+
+	protected Files $siblings;
+
+	public function __construct(
+		File $model
+	) {
+		parent::__construct($model);
+	}
+
+	public function breadcrumb(): array
+	{
+		$breadcrumb = [];
+		$parent     = $this->model->parent();
+
+		switch ($parent::CLASS_ALIAS) {
+			case 'user':
+				/** @var \Kirby\Cms\User $parent */
+				// The breadcrumb is not necessary
+				// on the account view
+				if ($parent->isLoggedIn() === false) {
+					$breadcrumb[] = [
+						'label' => $parent->username(),
+						'link'  => $parent->panel()->url(true)
+					];
+				}
+				break;
+			case 'page':
+				/** @var \Kirby\Cms\Page $parent */
+				$breadcrumb = $this->model->parents()->flip()->values(
+					fn ($parent) => [
+						'label' => $parent->title()->toString(),
+						'link'  => $parent->panel()->url(true),
+					]
+				);
+		}
```

Comment:
This part makes me wonder if we should have a UserFileViewController, SiteFileViewController and a PageFileViewController to handle different breadcrumbs and maybe potentially more differences later.

Reply:
Added it